### PR TITLE
[MASTER] feat: update the test suites

### PR DESCRIPTION
**Pull request by jhujhul**
This PR updates the test suites from the `tests` branch.

I use it to be sure the implementation from the candidates are ok.

It uses the examples from the README as a test case.
It also tests the following cases:
- if there're more than 2 duplicates: some implementations from the candidates work only with 2 duplicates
- using a different key than `'id'`
- test the array inputs are not mutated

### CODE REVIEW

File: __tests__/suites/mergeBy.js [hunk 1]
```diff
@@ -1,45 +1,135 @@
 const mergeBy = require('../../mergeBy');
 
 describe('mergeBy', () => {
-  it('should merge an array of objects using the provided key', () => {
+  it('should merge two arrays of objects using a specified key', () => {
```
Comment (jhujhul) [inline __tests__/suites/mergeBy.js:4]: That's the example from the README

**Review by ivanpigeaire — APPROVED** [review]
I'm not sure how/when we should use those however ? If we see there is a problem with an implementation should we just provide those tests and let the candidate have a look or can't we just challenge the problematic parts orally ?

**Comment by jhujhul** [comment]
> I'm not sure how/when we should use those however ? If we see there is a problem with an implementation should we just provide those tests and let the candidate have a look or can't we just challenge the problematic parts orally ?

How I think I'll use them:
When a candidate finishes an implementation which seems fine to me, I'll run those tests locally without telling the candidate, to double check the implementation is ok. If all tests pass, then all good. If no, it means I missed something. In that case, I can either challenge the problematic part orally, or provide the failing test and ask the candidate to have a look at it.
WDTY ?

**Comment by ivanpigeaire** [comment]
@jhujhul seems fine, could be useful indeed when the candidate comes up with a very weird/unreadable solution indeed

**Pull request merged**